inc/RestCommand.php

```diff
@@ -80,25 +86,32 @@ public function generate_items( $args, $assoc_args ) {
 		unset( $assoc_args['format'] );
 
 		$notify = false;
-		if ( 'progress' === $format ) {
+		if ( ( 'progress' === $format ) && !Utils\get_flag_value( $assoc_args, 'porcelain' ) ) {
 			$notify = \WP_CLI\Utils\make_progress_bar( 'Generating items', $count );
 		}
 
 		for ( $i = 0; $i < $count; $i++ ) {
-
 			list( $status, $body ) = $this->do_request( 'POST', $this->get_base_route(), $assoc_args );
-
-			if ( 'progress' === $format ) {
-				$notify->tick();
-			} else if ( 'ids' === $format ) {
-				echo $body['id'];
-				if ( $i < $count - 1 ) {
-					echo ' ';
+			if ( $status < 400 ) {
+				if ( Utils\get_flag_value( $assoc_args, 'porcelain' ) ) {
+					WP_CLI::line( $body['id'] );
+				} else if ( 'progress' === $format ) {
+					$notify->tick();
+				} else if ( 'ids' === $format ) {
+					echo $body['id'];
+					if ( $i < $count - 1 ) {
+						echo ' ';
+					}
+				}
+			} else {
+				if ( Utils\get_flag_value( $assoc_args, 'porcelain' ) ) {
+					return;
 				}
+				WP_CLI::error( 'Could not complete request.' );
```

Comment:
error codes now set in exit code and/or error text. Put error code at end of line without trailing period to make it easier for parsing.